lightstep/resource_metric_dashboard.go

```diff
@@ -890,6 +955,92 @@ func (p *resourceUnifiedDashboardImp) resourceUnifiedDashboardImport(ctx context
 
 }
 
+// buildServiceHealthPanels transforms service_health_panels from the TF resource
+// into panels for the API request.
+func buildServiceHealthPanels(serviceHealthPanelsIn []interface{}) ([]client.Panel, error) {
+	var serviceHealthPanels []client.Panel
+
+	for _, s := range serviceHealthPanelsIn {
+		serviceHealthPanel, ok := s.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("bad format, %v", s)
+		}
+		p := client.Panel{
+			ID:       serviceHealthPanel["id"].(string),
+			Title:    serviceHealthPanel["name"].(string),
+			Position: buildPosition(serviceHealthPanel),
+		}
+		// N.B. panel_options are optional, so we don't return an error if not found
+		if panelOptions, ok := serviceHealthPanel["panel_options"].(map[string]interface{}); ok {
+			displayOptions := make(map[string]interface{})
+			if sortBy, ok := panelOptions["sort_by"]; ok {
+				displayOptions["sort_by"] = sortBy
+			}
+			if sortDirection, ok := panelOptions["sort_direction"]; ok {
+				displayOptions["sort_direction"] = sortDirection
+			}
+			if percentile, ok := panelOptions["percentile"]; ok {
+				displayOptions["percentile"] = percentile
+			}
+			if changeSince, ok := panelOptions["change_since"]; ok {
+				displayOptions["change_since"] = changeSince
+			}
```

Comment:
Worth extracting this logic into a function that can be used across both `build` and `set` functions?

Something like:

```
func assignPanelOptions(from map[string]interface{}, to map[string]interface{}) {
	if sortBy, ok := from["sort_by"]; ok {
		to["sort_by"] = sortBy
	}
	if sortDirection, ok := from["sort_direction"]; ok {
		to["sort_direction"] = sortDirection
	}
	if percentile, ok := from["percentile"]; ok {
		to["percentile"] = percentile
	}
	if changeSince, ok := from["change_since"]; ok {
		to["change_since"] = changeSince
	}
}
````

Reply:
Reading this function made me realize: we can probably just use the interface _as is_. i.e. skip the conversion entirely.

Reply:
Actually we may want to explicitly do this. We might add a new display type in the publicapi, and we won't want to return that automatically in the publicapi -> TF resource conversion.